source/slang/slang-ir-liveness.cpp

```diff
@@ -875,10 +945,30 @@ void LivenessContext::_findInstRunsForBlocks()
                 }
             }
             SLANG_ASSERT(dst == m_instRuns.end());
+        }   
+
+        SLANG_ASSERT(blockInfo->runCount == blockInfo->instCount);
+
+        // Special case the terminator - we allow a return that accesses the root 
+        // to be added to the run. 
+        //
+        // TODO(JS): We might want this behavior to be switchable with an option.
+        // If we don't add the terminator, everything else will behave correctly with regard
+        // adding live range end markers.
+        {
+            auto terminator = block->getTerminator();
+            if (_isAccessTerminator(terminator))
+            {
+                m_instRuns.add(terminator);
+                blockInfo->runCount++;
+            }
         }
```

Comment:
Thinking more, I'm not clear how this scenario would arise in a way that matters. Wouldn't this only come up if a function was trying to return a pointer into stack-allocated memory?

Reply:
You can see it in the output. Without it you end up with a load into a temporary and end, and then a return of that temporary.